src/query/service/src/interpreters/interpreter_insert_v2.rs

```diff
@@ -197,10 +228,18 @@ impl InsertInterpreterV2 {
                             start.elapsed().as_secs()
                         );
                         table
-                            .commit_insertion(ctx, append_entries, overwrite)
+                            .commit_insertion(ctx.clone(), append_entries, overwrite)
                             .await?;
 
-                        // TODO:(everpcpc) purge copied files
+                        if stage_info.copy_options.purge {
```

Comment:
only pure when success?

Reply:
this whole block seems all under success condition?

Reply:
yes, I means do we need co consider to pure when fail